inc/css/class-block-frontend.php

```diff
@@ -0,0 +1,433 @@
+				if ( sizeof( $font['fontvariant'] ) > 0 ) {
+					$item .= ':' . implode( ',', $font['fontvariant'] );
+				}
+				array_push( $fonts, $item );
+			}
+
+			wp_enqueue_style( 'themeisle-gutenberg-google-fonts', '//fonts.googleapis.com/css?family=' . implode( '|', $fonts ) );
+		}
+	}
+
+	/**
+	 * Method to define hooks needed.
+	 * 
+	 * @since   1.2.5
+	 * @access  public
+	 */
+	public function enqueue_google_fonts_backward() {
+		if ( $this->has_fonts ) {
+			return;
+		}
+
+		$fonts = array();
+
+		if ( sizeof( self::$google_fonts ) > 0 ) {
+			foreach( self::$google_fonts as $font ) {
+				$item = str_replace( ' ', '+', $font['fontfamily'] );
+				if ( sizeof( $font['fontvariant'] ) > 0 ) {
+					$item .= ':' . implode( ',', $font['fontvariant'] );
+				}
+				array_push( $fonts, $item );
+			}
+
+			wp_enqueue_style( 'themeisle-gutenberg-google-fonts', '//fonts.googleapis.com/css?family=' . implode( '|', $fonts ) );
+		}
+	}
+
+	/**
+	 * Get Google Fonts for Reusable Blocks
+	 * 
+	 * @since   1.2.5
+	 * @access  public
+	 */
+	public function enqueue_reusable_fonts( $blocks ) {
+		foreach ( $blocks as $block ) {
+			if ( $block['blockName'] === 'core/block' && ! empty( $block['attrs']['ref'] ) ) {
+				$this->enqueue_google_fonts( $block['attrs']['ref'] );
+			}
+
+			if ( isset( $block['innerBlocks'] ) && ! empty( $block['innerBlocks'] ) && is_array( $block['innerBlocks'] ) ) {
+				$this->enqueue_reusable_fonts( $block['innerBlocks'] );
+			}
+		}
+	}
+
+	/**
+	 * Render server-side CSS
+	 * 
+	 * @since   1.2.5
+	 * @access  public
+	 */
+	public function render_post_css() {
+		if ( is_singular() ) {
+			return $this->enqueue_styles();
+		}
+
+		add_filter( 'the_content', function( $content ) {
+			if ( $this->has_excerpt ) {
+				return $content;
+			}
+
+			$post_id = get_the_ID();
+
+			$this->enqueue_styles( $post_id, true );
+
+			return $content;
+		} );
+	}
+
+	/**
+	 * Enqueue CSS file
+	 * 
+	 * @since   1.2.5
+	 * @access  public
+	 */
+	public function enqueue_styles( $post_id = '', $footer = false ) {
+		$post_id = $post_id ? $post_id : get_the_ID();
+		$location = 'wp_head';
+
+		if ( function_exists( 'has_blocks' ) && has_blocks( $post_id ) ) {
+			$file_name = get_post_meta( $post_id, '_themeisle_gutenberg_block_stylesheet', true );
+
+			if ( $footer ) {
+				$location = 'wp_footer';
+			}
+
+			if ( empty( $file_name ) || is_preview() ) {
+				return add_action( $location, function() use( $post_id ) {
+					return $this->get_post_css( $post_id );
+				} );
+			}
+
+			$wp_upload_dir = wp_upload_dir( null, false );
+			$basedir = $wp_upload_dir['basedir'] . '/themeisle-gutenberg/';
+			$baseurl = $wp_upload_dir['baseurl'] . '/themeisle-gutenberg/';
+			$file_path = $basedir . $file_name . '.css';
+			$file_url = $baseurl . $file_name . '.css';
+
+			if ( ! file_exists( $file_path ) ) {
+				return add_action( $location, function() use( $post_id ) {
+					return $this->get_post_css( $post_id );
```

Comment:
also, we should generate and save the css file as well, if it not exists.

Reply:
@selul I'm thinking of not doing it with this release. While everything is well tested, we never know what will happen once the plugin is released in the wild. So to avoid issues, maybe we can take this step in the next release? What do you think?

Reply:
Yes, sounds fair. 

So this release we are aggregating the style into a single style tag and we can tackle the saving of it to separate CSS files in the next release ? 
If so, we can remove the file saving logic handling from this release, or at least comment the code.

Reply:
@selul Not that. I meant to not save CSS into file and meta for existing posts.

Reply:
ah, fair enough